internal/language/language_test.go

```diff
@@ -116,3 +116,59 @@ func TestRecognizedSupporteds(t *testing.T) {
 		assert.NotEmpty(t, l.Executable().Name())
 	}
 }
+
+func TestMakeByNameAndVersion(t *testing.T) {
+	type args struct {
+		name    string
+		version string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    Language
+		wantErr bool
+	}{
+		{
+			"Valid Python3 version",
+			args{"python", "3.6.6"},
+			Python3,
+			false,
+		},
+		{
+			"Valid Python2 version",
+			args{"python", "2.7.18"},
+			Python2,
+			false,
+		},
+		{
+			"Valid Python2 invalid patch",
+			args{"python", "2.7.18.1"},
+			Python2,
+			false,
+		},
+		{
+			"Invalid version",
+			args{"python", ""},
+			Unknown,
+			true,
+		},
+		{
+			"Valid Perl version",
+			args{"perl", "5.28.1"},
+			Perl,
+			false,
```

Comment:
Please assert the versions also

Reply:
The version isn't returned from the `MakeByNameAndVersion` function so I'm not sure how to assert that. It's implicitly asserted in the Python tests but not for Perl